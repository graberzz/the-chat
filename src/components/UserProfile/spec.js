import React from 'react';
import { shallow } from 'enzyme'; 
import { UserProfile, ButtonFlex } from './';

describe('<UserProfile />', () => {
  let userProfile;
  
  const renderUserProfile = ({ login = f => f, logout = f => f, user = null } = {}) =>
    shallow(<UserProfile
      login={login}
      logout={logout}
      user={user}
    />);

  it('matches logged in snapshot', () => {
    userProfile = renderUserProfile();
    expect(userProfile).toMatchSnapshot();
  });

  it('matches logged out snapshot', () => {
    userProfile = renderUserProfile({
      user: {
        username: 'user',
        avatar: 'hhtps',
      },
    });
    expect(userProfile).toMatchSnapshot();
  });

  it('fires login', () => {
    const mockLogin = jest.fn();

    userProfile = renderUserProfile({ login: mockLogin });
    userProfile.find('ButtonFlex').simulate('click');
    expect(mockLogin).toBeCalled();
  });

  it('fires logout', () => {
    const mockLogout = jest.fn();
    const user = {
      username: 'user',
      avatar: 'link',
    };

    userProfile = renderUserProfile({
      logout: mockLogout,
      user: {
        username: 'user',
        avatar: 'url',
      },
    });
    userProfile.find('ButtonFlex').simulate('click');
    expect(mockLogout).toBeCalled();
  });
})