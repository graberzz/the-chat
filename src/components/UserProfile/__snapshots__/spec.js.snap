// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserProfile /> matches logged in snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    login={[Function]}
    logout={[Function]}
    user={null}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <styled.img
          alt="Google"
          src="/google.png"
/>,
        "Login",
      ],
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "alt": "Google",
          "src": "/google.png",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "Login",
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.img
            alt="Google"
            src="/google.png"
/>,
          "Login",
        ],
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "alt": "Google",
            "src": "/google.png",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "Login",
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<UserProfile /> matches logged out snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    login={[Function]}
    logout={[Function]}
    user={
        Object {
            "avatar": "hhtps",
            "username": "user",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <styled.img
          alt="user"
          src="hhtps"
/>,
        "Logout",
      ],
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "alt": "user",
          "src": "hhtps",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "Logout",
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.img
            alt="user"
            src="hhtps"
/>,
          "Logout",
        ],
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "alt": "user",
            "src": "hhtps",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "Logout",
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
